pkg/tsdb/cloudwatch/services/list_metrics.go

```diff
@@ -131,11 +132,11 @@ func setDimensionFilter(input *cloudwatch.ListMetricsInput, dimensionFilter []*r
 	}
 }
 
-func setAccount(input *cloudwatch.ListMetricsInput, r *resources.ResourceRequest) {
-	if r != nil && r.AccountId != nil {
+func (l *ListMetricsService) setAccount(input *cloudwatch.ListMetricsInput, r *resources.ResourceRequest) {
+	if l.isCrossAccountEnabled {
```

Comment:
Previously, we were only setting `IncludeLinkedAccounts` if an accountID was set, which seems incorrect to me since we always set it for the log groups query if `isCrossAccountEnabled` is true, but I can undo this if there was a reason I didn't know about.

Reply:
So here we could run into a case where isCrossAccountEnabled = true but have AccountID = nil -> panic 

In the log groups query, maybe it is ensured that the AccountID is present? I don't really remember off the top of my head.

Does this change need to be part of this PR?

Reply:
Reread it and I was misunderstanding what happened for LogGroup queries, undone!